### PR TITLE
Updates to all components enforced by restructuring of `contracts`.

### PR DESCRIPTION
**Pull request by jplodge-pro**
This PR is targeted at the `restructure-contracts` branch.

### DONE
- [x] Renamed everything `kyc` to be `whiteList`.
- [x] Use `CredbullWhiteListProvider` in place of `CredbullKYCProvider`.
- [x] Corrected all usages of the changed `Params` tree structure.
- [x] Use `VaultFactory` in place of `CredbullVaultFactory`
- [x] Use `SimpleUSDC`in place of `MockStablecoin`
- [x] Use `SimpleToken` in place of `MockToken` 
- [x] In `api`, do a proper migration of Supabase Tables, Indices, Types etc for the changeover to 'whiteList' over 'kyc'.

### CODE REVIEW

File: packages/api/src/modules/accounts/whiteList.service.ts [hunk 1]
```diff
@@ -89,20 +89,20 @@ export class KycService {
...
       .select();
   }
 
   private async checkOnChain(
-    kycProviders: Tables<'vault_entities'>[],
+    whiteListProviders: Tables<'vault_entities'>[],
     address: string,
   ): Promise<ServiceResponse<boolean>> {
     const errors = [];
-    let status = false;
```
Comment (jplodge-pro) [inline packages/api/src/modules/accounts/whiteList.service.ts:102]: This starting value of `false` means that the below loop will never return a true. This is a bug, I believe and has been fixed.

File: packages/app/src/utils/rsc.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
 export const getSegment = (props: any) => {
-  return props.children.props.childProp.segment;
+  console.log('SEGMENT=', props.children?.props?.childProp?.segment);
+  // FIX (JL,2024-07-05): This deref chain always results in `undefined`. How to fix?
+  return props.children?.props?.childProp?.segment;
```
Comment (jplodge-pro) [inline packages/app/src/utils/rsc.ts:4]: I don't understand this problem, other than it seems to have something to do with Routes and Segments. This remains an issue,

File: packages/api/package.json [hunk 3]
```diff
@@ -66,7 +66,7 @@
     "@types/compression": "^1.7.5",
     "@types/express": "^4.17.21",
     "@types/lodash": "^4.14.202",
-    "@types/node": "^20.10.5",
+    "@types/node": "^20.14.9",
```
Comment (Krishnakumarskr) [inline packages/api/package.json:69]: Do we need to change this node version???

File: packages/api/src/modules/accounts/whiteList.service.ts (hunk 1 above)
Comment (Krishnakumarskr) [reply packages/api/src/modules/accounts/whiteList.service.ts:102]: Confirming this is a bug!

For the fix, On line 113, the NOT(!) operator should be removed.
The loop should break if the status is once set to true after the line `status = status && data`

File: packages/api/package.json (hunk 3 above)
Comment (jplodge-pro) [reply packages/api/package.json:69]: Need to? No, it was one of the things I tried to improve things. I did not think it was significant. I will revert.

File: packages/api/src/modules/accounts/whiteList.service.ts (hunk 1 above)
Comment (jplodge-pro) [reply packages/api/src/modules/accounts/whiteList.service.ts:102]: So you are saying ANY White List Provider can whitelist an address?
Actually, it sounds logical when you put it that way. Thanks, will fix.

**Pull request merged**